coderd/provisionerjobs.go

```diff
@@ -343,3 +338,46 @@ func convertProvisionerJob(provisionerJob database.ProvisionerJob) codersdk.Prov
 func provisionerJobLogsChannel(jobID uuid.UUID) string {
 	return fmt.Sprintf("provisioner-log-logs:%s", jobID)
 }
+
+// provisionerJobLogsMessage is the message type published on the provisionerJobLogsChannel() channel
+type provisionerJobLogsMessage struct {
+	EndOfLogs bool                         `json:"end_of_logs,omitempty"`
+	Logs      []database.ProvisionerJobLog `json:"logs,omitempty"`
+}
+
+func (api *API) followLogs(jobID uuid.UUID) (chan database.ProvisionerJobLog, func(), error) {
+	bufferedLogs := make(chan database.ProvisionerJobLog, 128)
```

Comment:
One little thing to help logger fields maybe.

```golang
logger := api.Logger.With(slog.F("job_id", jobID))
```

Then just `logger.Warn()` and `logger.Debug()` later. The field is always present.